src/disassembler/disassembler.tsx

```diff
@@ -6,17 +6,23 @@ interface Instruction {
 
 export default class RiscVDisassembler {
     /**
-     * Disassemble an instruction
-     * @param instr - The hex instruction to disassemble
+     * Disassemble a hex or binary instruction
+     * @param instr - The instruction to disassemble
+     * @param radix - An integer between 2 and 36 specifying the base to use for representing numeric values
      */
-    static disassembleInstruction(instr: string): string {
-        if (instr.length !== 8) return "Please ensure that the instruction is 32 bits!";
+    static disassembleInstruction(instr: string, radix: number): string {
+        if (radix === 2) {
```

Comment:
Could this be changed to something like `if (radix * instr.length !== 32)`? That way we if someone for some reason wanted to use, like, octal or something, they can just put in radix 8, and not have to come back and edit this file again

Reply:
> Could this be changed to something like `if (log2(radix) * instr.length !== 32)`? That way we if someone for some reason wanted to use, like, octal or something, they can just put in radix 8, and not have to come back and edit this file again

Yep